pkgs/by-name/ca/cargo-tauri/package.nix

```
@@ -4,39 +4,41 @@
   callPackage,
   rustPlatform,
   fetchFromGitHub,
+  cargo-tauri,
   darwin,
-  gtk3,
-  libsoup,
+  gtk4,
+  nix-update-script,
   openssl,
   pkg-config,
-  webkitgtk,
+  testers,
+  webkitgtk_4_1,
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "tauri";
-  version = "1.7.1-unstable-2024-08-16";
+  version = "2.0.1";
 
   src = fetchFromGitHub {
     owner = "tauri-apps";
     repo = "tauri";
-    rev = "2b61447dfc167ec11724f99671bf9e2de0bf6768";
-    hash = "sha256-gKG7olZuTCkW+SKI3FVZqgS6Pp5hFemRJshdma8rpyg=";
+    rev = "refs/tags/tauri-v${version}";
+    hash = "sha256-kHzpx1o894bo4Ud3D1lbB+pZahflF/o2bb2EDJUI148=";
   };
 
-  # Manually specify the sourceRoot since this crate depends on other crates in the workspace. Relevant info at
-  # https://discourse.nixos.org/t/difficulty-using-buildrustpackage-with-a-src-containing-multiple-cargo-workspaces/10202
-  sourceRoot = "${src.name}/tooling/cli";
-
-  cargoHash = "sha256-VXg/dAhwPTSrLwJm8HNzAi/sVF9RqgpHIF3PZe1LjSA=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "schemars_derive-0.8.21" = "sha256-AmxBKZXm2Eb+w8/hLQWTol5f22uP8UqaIh+LVLbS20g=";
+    };
+  };
```

```suggestion
cargoHash = "hash"
```

as this is not really needed as [`buildRustPackage` uses cargoLock to provide a hash](https://github.com/NixOS/nixpkgs/blob/5966581aa04be7eff830b9e1457d56dc70a0b798/pkgs/build-support/rust/build-rust-package/default.nix#L62-L79) instead to make it easier for your self use `cargohash`.

---

https://github.com/NixOS/nixpkgs/blob/5966581aa04be7eff830b9e1457d56dc70a0b798/pkgs/build-support/rust/import-cargo-lock.nix

---

I believe you're misunderstanding some things here. From the "Vendoring dependencies" [section](https://nixos.org/manual/nixpkgs/unstable/#vendoring-of-dependencies) section of the Rust reference:

> If the Cargo.lock file includes git dependencies, then their output hashes need to be specified since they are not available through the lock file.

We can't use `fetchCargoTarball`/`cargoHash` to do this at all

---

I guess it makes sense as nix is ment to be reproducible.